src/parallel/parallel_ghost_sync.C

```diff
@@ -88,4 +88,32 @@ void SyncSubdomainIds::act_on_data (const std::vector<dof_id_type> & ids,
     }
 }
 
+SyncElementIntegers::SyncElementIntegers(MeshBase & m, const std::string & integer_name)
+  : mesh(m), ind(m.get_elem_integer_index(integer_name))
+{}
+
+void SyncElementIntegers::gather_data(const std::vector<dof_id_type> & ids,
+                                      std::vector<dof_id_type> & ids_out) const
+{
+  ids_out.reserve(ids.size());
+
+  for (const auto & id : ids)
+    {
+      Elem * elem = mesh.query_elem_ptr(id);
+      libmesh_assert(elem);
+      ids_out.push_back(elem->get_extra_integer(ind));
+    }
+}
+
+void SyncElementIntegers::act_on_data(const std::vector<dof_id_type> & ids,
+                                      const std::vector<dof_id_type> & integer_ids) const
+{
+  for (auto i : index_range(ids))
+    {
+      Elem * elem = mesh.query_elem_ptr(ids[i]);
```

Comment:
Let's just make this `mesh.elem_ref(ids[i])`, which lets you delete a then-redundant assert and a layer of indirection below?